lms/views/admin/application_instance.py

```diff
@@ -13,6 +14,35 @@
 from lms.views.admin import flash_validation
 from lms.views.admin._schemas import EmptyStringInt
 
+# Helper to declare settings as secret
+AES_SECRET = object()
+APPLICATION_INSTANCE_SETTINGS = {
+    ("blackboard", "files_enabled"): asbool,
+    ("blackboard", "groups_enabled"): asbool,
+    ("canvas", "sections_enabled"): asbool,
+    ("canvas", "groups_enabled"): asbool,
+    ("desire2learn", "client_id"): str,
+    ("desire2learn", "client_secret"): AES_SECRET,
+    ("desire2learn", "groups_enabled"): asbool,
+    ("desire2learn", "files_enabled"): asbool,
+    ("desire2learn", "create_line_item"): asbool,
+    ("microsoft_onedrive", "files_enabled"): asbool,
+    ("vitalsource", "enabled"): asbool,
+    ("vitalsource", "user_lti_param"): str,
+    ("vitalsource", "user_lti_pattern"): str,
+    ("vitalsource", "api_key"): str,
+    ("vitalsource", "disable_licence_check"): asbool,
+    ("jstor", "enabled"): asbool,
+    ("jstor", "site_code"): str,
+    ("hypothesis", "notes"): str,
+}
+
+APPLICATION_INSTANCE_SETTINGS_COLUMNS = tuple(
```

Comment:
COLUMNS? Maybe we should mention "for search" / "filters" or something along those lines.